Conservative validation in `kafka::group`

mmaslankaprv opened the pull request:
### Term check

Added a check validating if partition term change when group state is
being validated. It is perfectly possible for the same node to become a
leader for the consumer group partition after it stepped down. Therefore
checking if a current replica is the leader is not enough. Term check
validates if current group state is fully up to date even if the replica
is still the leader.

### Moving offset backward

When consumer requests committed offset to be updated to smaller value
that a current one it is suspicious. It is better to have this reported
in the logs to make debugging easier.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.1.x
- [x] v24.3.x
- [x] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

mmaslankaprv committed:
k/group: log information about committed offset moving backward

When consumer requests committed offset to be updated to smaller value
that a current one it is suspicious. It is better to have this reported
in the logs to make debugging easier.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
k/group: added a conservative term check in group_manager

Added a check validating if partition term change when group state is
being validated. It is perfectly possible for the same node to become a
leader for the consumer group partition after it stepped down. Therefore
checking if a current replica is the leader is not enough. Term check
validates if current group state is fully up to date even if the replica
is still the leader.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/63594-->
<!--550de620-2b66-4649-977b-52db88515dcc-->
<!--65f1bd88-4976-4fa9-89d2-63a2a7b8a231-->
<!--5156e883-8457-43c4-bb8a-2656dbd1b66f-->
<!--title-->
## Retry command for [Build#63594](https://buildkite.com/redpanda/redpanda/builds/63594)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/shard_placement_test.py::ShardPlacementTest.test_upgrade
tests/rptest/tests/consumer_offsets_consistency_test.py::ConsumerOffsetsConsistencyTest.test_flipping_leadership
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_dropped_column_no_collision@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"produce_mode":"avro","query_engine":"spark"}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--d09d0919-baf5-459a-ae01-41ddb870e3c2-->
<!--TEST RESULTS-->
<!--c9f9d5dc-99cd-4b0b-a65c-87f7df430cef-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63594">build#63594</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.kafka_streams_test.KafkaStreamsAnomalyDetection.test_kafka_streams | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.compatibility.sarama_test.SaramaTest.test_sarama_consumergroup | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99e-e368-44c7-9a78-59888718f602 | FLAKY | 1/2 |
| rptest.tests.consumer_group_recovery_test.ConsumerOffsetsRecoveryTest.test_consumer_offsets_partition_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.consumer_group_test.ConsumerGroupTest.test_dead_group_recovery.static_members=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.consumer_group_test.ConsumerGroupTest.test_group_lag_metrics | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/13 |
| rptest.tests.consumer_offsets_consistency_test.ConsumerOffsetsConsistencyTest.test_flipping_leadership | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FAIL | 0/20 |
| rptest.tests.cpu_profiler_admin_api_test.CPUProfilerAdminAPITest.test_get_cpu_profile_with_override | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.in.stage.executed.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.in.stage.executing.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.in.stage.prepared.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/7 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.in.stage.preparing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/12 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.in.stage.preparing.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/9 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.out.stage.executing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.out.stage.prepared.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=False.params=.cancellation.dir.out.stage.preparing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/5 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.executed.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.executed.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.executing.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.prepared.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.prepared.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.out.stage.preparing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/4 |
| rptest.tests.datalake.3rdparty_maintenance_test.Datalake3rdPartyMaintenanceTest.test_e2e_basic.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.datalake.3rdparty_maintenance_test.Datalake3rdPartyMaintenanceTest.test_e2e_basic.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.datalake.3rdparty_maintenance_test.Datalake3rdPartyMaintenanceTest.test_e2e_basic.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.datalake.custom_partitioning_test.DatalakeCustomPartitioningTest.test_spec_evolution_rpcn.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_dlq_test.DatalakeDLQTest.test_dlq_table_for_invalid_records.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_dlq_test.DatalakeDLQTest.test_dlq_table_for_invalid_records.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_dlq_test.DatalakeDLQTest.test_dlq_table_for_invalid_records.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_verifier_test.DatalakeVerifierTest.test_detecting_duplicates.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.datalake.datalake_verifier_test.DatalakeVerifierTest.test_detecting_gap_in_offset_sequence.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/6 |
| rptest.tests.datalake.partition_movement_test.PartitionMovementTest.test_cross_core_movements.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FAIL | 0/20 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/11 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/19 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.AVRO.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO2.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/12 |
| rptest.tests.datalake.schema_evolution_test.SchemaEvolutionE2ETests.test_dropped_column_no_collision.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.produce_mode=ProducerType.PROTO3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/5 |
| rptest.tests.datalake.transactions_test.DatalakeTransactionTests.test_with_transactions.cloud_storage_type=CloudStorageType.S3.compaction=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_concurrent_truncations.cloud_storage_enabled=False.truncate_point=start_offset | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/4 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_concurrent_truncations.cloud_storage_enabled=True.truncate_point=random_offset | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_concurrent_truncations.cloud_storage_enabled=True.truncate_point=start_offset | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/4 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_concurrent_truncations.cloud_storage_enabled=True.truncate_point=start_offset | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99e-e368-44c7-9a78-59888718f602 | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset_spillover.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_write.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.follower_fetching_test.FollowerFetchingTest.test_basic_follower_fetching.read_from_object_store=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.follower_fetching_test.FollowerFetchingTest.test_follower_fetching_with_maintenance_mode | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.follower_fetching_test.FollowerFetchingTest.test_with_leadership_transfers | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.group_membership_test.ListGroupsReplicationFactorTest.test_list_groups | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.idempotency_test.IdempotencyWriteCachingTest.test_idempotent_producers_write_caching | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.log_segment_ms_test.SegmentMsTest.test_segment_not_rolling.use_alter_cfg=False.server_cfg=None.topic_cfg=None | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.mirror_maker_test.TestMirrorMakerService.test_consumer_group_mirroring.source_type=redpanda | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/5 |
| rptest.tests.mirror_maker_test.TestMirrorMakerService.test_simple_end_to_end.source_type=redpanda | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.nodes_decommissioning_test.NodesDecommissioningTest.test_decommissioning_finishes_after_manual_cancellation.delete_topic=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.nodes_decommissioning_test.NodesDecommissioningTest.test_recommissioning_do_not_stop_all_moves_node | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.offset_retention_test.OffsetRetentionTest.test_offset_expiration | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.pandaproxy_test.PandaProxyAutoAuthTest.test_create_consumer_validation | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.pandaproxy_test.PandaProxyAutoAuthTest.test_subscribe_consumer_validation | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.pandaproxy_test.PandaProxyBasicAuthTest.test_consumer | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.pandaproxy_test.PandaProxyConsumerGroupTest.test_moving_group_coordinator | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/13 |
| rptest.tests.pandaproxy_test.PandaProxyInvalidInputsTest.test_invalid_subscribe_consumer | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/5 |
| rptest.tests.pandaproxy_test.PandaProxyTest.test_consumer_group_binary_v2 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/9 |
| rptest.tests.pandaproxy_test.PandaProxyTest.test_create_consumer_validation | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/8 |
| rptest.tests.pandaproxy_test.PandaProxyTest.test_remove_consumer_validation | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/10 |
| rptest.tests.pandaproxy_test.PandaProxyTest.test_restart_http_proxy | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/10 |
| rptest.tests.pandaproxy_test.PandaProxyTest.test_subscribe_consumer_validation | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/13 |
| rptest.tests.partition_balancer_test.PartitionBalancerTest.test_maintenance_mode.kill_same_node=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99e-e368-44c7-9a78-59888718f602 | FLAKY | 1/2 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move.replication_factor=1.unclean_abort=True.recovery=no_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move.replication_factor=3.unclean_abort=False.recovery=no_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move.replication_factor=3.unclean_abort=True.recovery=no_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=1.unclean_abort=False.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/10 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=1.unclean_abort=True.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=1.unclean_abort=True.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99e-e368-44c7-9a78-59888718f602 | FLAKY | 1/2 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=3.unclean_abort=False.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/6 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=3.unclean_abort=True.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/11 |
| rptest.tests.partition_movement_upgrade_test.PartitionMovementUpgradeTest.test_basic_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.recovery_mode_test.RecoveryModeTest.test_rolling_restart | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.redpanda_multi_region_test.RedpandaNetworkDelayTest.test_basic_traffic_shaping | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.rpk_group_test.RpkGroupCommandsTest.test_group_describe | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.rpk_group_test.RpkGroupCommandsTest.test_group_list_and_delete | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/7 |
| rptest.tests.rpk_group_test.RpkGroupCommandsTest.test_group_seek_to_file | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/10 |
| rptest.tests.rpk_group_test.RpkGroupCommandsTest.test_group_seek_to_group | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.rpk_group_test.RpkGroupCommandsTest.test_group_seek_to_timestamp | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.rpk_topic_test.RpkToolTest.test_consume_as_group | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/5 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_adding_nodes_to_cluster.partition_count=1 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_adding_nodes_to_cluster.partition_count=20 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_on_demand_rebalancing.partition_count=20 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.Payload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_schema_id_validation.protocol=JSON.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_serde_client.protocol=PROTOBUF.client_type=Java.skip_known_types=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.RECORD_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/5 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.RECORD_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/4 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.Payload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.Payload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=JSON.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.RECORD_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=JSON.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.Payload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=PROTOBUF.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.RECORD_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=PROTOBUF.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/5 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=PROTOBUF.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.CompressiblePayload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_schema_id_validation.protocol=PROTOBUF.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.Payload.compression_type=CompressionTypes.NONE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/3 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_serde_client.protocol=PROTOBUF.client_type=Java.skip_known_types=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_serde_client.protocol=PROTOBUF.client_type=Python.skip_known_types=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.tests.services_self_test.KgoVerifierSelfTest.test_kgo_verifier | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99e-e368-44c7-9a78-59888718f602 | FLAKY | 1/2 |
| rptest.tests.shard_placement_test.ShardPlacementTest.test_core_count_change | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/7 |
| rptest.tests.shard_placement_test.ShardPlacementTest.test_node_join.disable_license=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9b-4fee-accc-123418dc4984 | FLAKY | 1/2 |
| rptest.tests.shard_placement_test.ShardPlacementTest.test_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FAIL | 0/20 |
| rptest.transactions.transactions_test.TransactionsTest.graceful_leadership_transfer_tx_coordinator_test | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.transactions.transactions_test.TransactionsTest.rejoin_member_test | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9c-4b32-8d82-0c8f4df951ad | FLAKY | 1/2 |
| rptest.transactions.transactions_test.TransactionsTest.simple_test | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.transactions.tx_atomic_produce_consume_test.TxAtomicProduceConsumeTest.test_basic_tx_consumer_transform_produce.with_failures=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
| rptest.transactions.tx_streams_test.TransactionsStreamsTest.consumer_offsets_retention_test | ducktape | https://buildkite.com/redpanda/redpanda/builds/63594#0195c99c-ec9a-48db-8073-2c4786bfc15c | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63651">build#63651</a></summary>


| test_id                                                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.compaction_test.CompactionTest.test_compaction.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63651#0195ccd6-17e5-4ac7-9c16-31f324d3d3e7 | FLAKY         | 1/2      |
| rptest.tests.datalake.compaction_test.CompactionTest.test_compaction.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP            | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63651#0195ccd6-17e6-4e16-940b-7128e4c99462 | FLAKY         | 1/2      |
| rptest.tests.datalake.datalake_e2e_test.DatalakeE2ETests.test_topic_lifecycle.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63651#0195ccd1-df4c-40a6-abce-d121e69ce5ef | FLAKY         | 1/2      |
| rptest.tests.datalake.recovery_mode_test.DatalakeRecoveryModeTest.test_recovery_mode.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63651#0195ccd6-17e5-4ac7-9c16-31f324d3d3e7 | FLAKY         | 1/2      |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic                                                                             | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63651#0195ccd6-17e3-4c4f-83a8-9a885a483bd1 | FLAKY         | 1/3      |
</details>
<!--/body-->

dotnwat commented on src/v/kafka/server/group_manager.cc:
```diff
@@ -1812,6 +1812,20 @@ error_code group_manager::validate_group_status(
+        if (it->second->partition->term() != it->second->term()) {
+            vlog(
+              cg_klog.info,
+              "Group {} operation {} sent to partition but leadership changed. "
+              "Group term: {} current term: {}",
+              group,
+              api,
+              ntp);
```
there are more `{}` than arguments in the format string.

dotnwat replied:
there are more `{}` than arguments in the format string.

dotnwat reviewed (commented):
* lgtm other than the format string comment.

* also, should we log it at warning level? not sure

i wonder if you could add a similar check in the `group_recovery_consumer`. that consumer keeps the last entry in the log, BUT, it gets to see all the entries, too. so it might be a good spot to catch any weird states?

mmaslankaprv committed:
k/recovery_consumer: log at info level when committed offset moves back

It is not common operation for the consumers to commit offset that is
smaller than the previous one. Logging with info level during consumer
group recovery when partition committed offset moves backward will give
us better visibility into potential problems with offset commits.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

pgellert commented on src/v/kafka/server/group_manager.cc:
```diff
@@ -1812,6 +1812,23 @@ error_code group_manager::validate_group_status(
+         * became a leader again.
+         */
+        if (it->second->partition->term() != it->second->term()) {
+            vlog(
+              cg_klog.info,
+              "Group {} operation {} for partition {} processed while term "
+              "changed. "
+              "Group term: {} current term: {}",
+              group,
+              api,
+              ntp,
+              it->second->term(),
+              it->second->partition->term());
+            return error_code::not_coordinator;
```
In this scenario of "the same node becomes a leader for the consumer group partition after it stepped down", how will `it->second->term()` eventually update? I assume there will be a notification callback from `register_leadership_notification` even if the leader didn't change only the term, which will update the group's term. Is that correct?

mmaslankaprv replied:
exactly as you described that, there will be two notifications

pgellert replied:
Thanks

pgellert reviewed (approved):
lgtm

vbotbuildovich commented:
/backport v25.1.x

vbotbuildovich commented:
/backport v24.3.x

vbotbuildovich commented:
/backport v24.2.x

vbotbuildovich commented:
Failed to create a backport PR to v24.3.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25533-v24.3.x-586 remotes/upstream/v24.3.x
git cherry-pick -x d7dfd9545e 09ac67b72e d8e6d47f57
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/14061379292).

vbotbuildovich commented:
Failed to create a backport PR to v24.2.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25533-v24.2.x-356 remotes/upstream/v24.2.x
git cherry-pick -x d7dfd9545e 09ac67b72e d8e6d47f57
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/14061379756).
